quipucords/scanner/satellite/api.py

```diff
@@ -126,7 +130,53 @@ def _prepare_host_request_options(self):
         }
         return request_options
 
-    def _process_hosts_using_multiprocessing(
+    def _prepare_and_process_hosts(
+        self,
+        hosts: Iterable[dict],
+        request_host_details: Callable,
+        process_results: Callable,
+        manager_interrupt: Value = None,
+    ):
+        if settings.QPC_ENABLE_CELERY_SCAN_MANAGER:
+            self._prepare_and_process_hosts_using_celery(
+                hosts, request_host_details, process_results
+            )
+        else:
+            self._prepare_and_process_hosts_using_multiprocessing(
+                hosts, request_host_details, process_results, manager_interrupt
+            )
+
+    def _prepare_and_process_hosts_using_celery(
+        self, hosts: Iterable[dict], request_host_details, process_results: Callable
+    ):
+        """Prepare and process hosts using Celery tasks.
+
+        :param hosts: iterable of host dicts
+        :param request_host_details: API version-specific function to get host details,
+            must also be a registered Celery task
+        :param process_results: API version-specific function to process results
+        """
+        all_prepared_hosts = self.prepare_host(hosts, ids_only=True)
+        # Note: the second argument for `chunks`
+        # pylint: disable=fixme
+        # TODO should we use something other than self.max_concurrency to drive this?
```

Comment:
do we even need to limit? the client could have handle rate limit errors (assuming satellite can emit those).

IMO max concurrency is only relevant to network scan

Reply:
🪓'd